Dev

VasckMe opened the pull request:
PR from dev to main branch

VasckMe committed:
Sign In and Sign Up intertefaces created

VasckMe committed:
Verifycation interface created

VasckMe committed:
Added ScrollView to Sign Up Interface

VasckMe committed:
Created Outlets and IBActions in Sign In Controller

VasckMe committed:
Created Outlets and IBActions for Sign Up Controller

VasckMe committed:
Small fixes, nevermind

VasckMe committed:
Created Outlets and IBActions for Verifying Controller

VasckMe committed:
Created Welcome interface

VasckMe committed:
Created IBAction in Welcome Controller

VasckMe committed:
Created enumeration model for password difficult

VasckMe committed:
Created account model for users

VasckMe committed:
Removed useless senders from IBActions, added some IBActions, small fixes

VasckMe committed:
Added show password button and IBAction for checking password

VasckMe committed:
Added "database" collection for accounts and small fixes in project.pbxproj

VasckMe committed:
Created authorization for Sign In interface

VasckMe committed:
Added working function "difficult of password" on Sign Up Interface

VasckMe committed:
Added:
- unwind from Welcome View Contrtoller to Sign In Root Controller
- button Sign In on Sign Up Interface
- IBActions: Sign In on Sign Up Interface, SignUp button to Verifying View Controller

VasckMe committed:
Sign Up working interface finished with sending data finished!

VasckMe committed:
Created logic for Verify VC ( without service :c )

VasckMe committed:
- button Sign In will be enabled only after correct email and password
- added tab bar controller after Sign In

VasckMe committed:
Added function to find index of user in database

VasckMe committed:
Global update in code:
- changed architecture of the project
- Added VerificationService
- Changed some names
- Removed some functions

VasckMe committed:
test1

VasckMe committed:
Revert "test1"

This reverts commit 1d1dd797759e3df05f8d95dd0dde315dad115cbf.

VasckMe committed:
Fixed SignIn button

VasckMe committed:
Created tab bar controller

VasckMe committed:
Small changes

VasckMe committed:
Fix

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/ProfileViewController.swift:
```diff
@@ -0,0 +1,35 @@
+class ProfileViewController: MainViewController {
+
+    @IBOutlet weak var nameLbl: UILabel!
+    @IBOutlet weak var emailLbl: UILabel!
```
emailLbl -> emailLabel

VasckMe replied:
Fixed

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/ProfileViewController.swift:
```diff
@@ -0,0 +1,35 @@
+
+import UIKit
+
+class ProfileViewController: MainViewController {
```
why not final?

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    
+    //MARK: Private properties
+    private var isValidPassword = false { didSet {isValidateUser()}}
+    private var isValidEmail = false { didSet {isValidateUser()}}
```
not in one line

VasckMe replied:
Ok

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    @IBOutlet private var passTextField: UITextField!
+    @IBOutlet private var wrongPassLbl: UILabel!
+    @IBOutlet private var showPassLbl: UIButton!
+    @IBOutlet private weak var signInButtonOutlet: UIButton!
```
why this button is weak, not previous one is not?

VasckMe replied:
I don't  know, but I have fixed it

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    //MARK: IBActions
+    @IBAction func checkEmail() {
```
IBActions also should be private

VasckMe replied:
Encapsulation fixed

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,95 @@
+        if let email = emailTextField.text,
+           !email.isEmpty {
+            isValidEmail = VerificationService.isValidEmailAddress(emailAddressString: email)
+        }
```
Update style:
```
if 
            let email = emailTextField.text,
            !email.isEmpty 
{
             isValidEmail = VerificationService.isValidEmailAddress(emailAddressString: email)
}
```

VasckMe replied:
Updated

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,95 @@
+        wrongEmailLbl.isHidden = isValidEmail
+    }
+    @IBAction func checkPassword() {
+        if let password = passTextField.text,
```
update style, check the previous comment

VasckMe replied:
I didn't know if style, have already fixed that

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    @IBAction func showPassButton() {
+        switch passTextField.isSecureTextEntry {
```
For Bool use `if`, not `switch`

VasckMe replied:
Fixed =)

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,95 @@
+
+    @IBAction func signInButton() {
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        guard let tabBarVC = storyboard.instantiateViewController(withIdentifier: "TabBarVC") as? TabBarViewController else { return }
```
Style:
`guard 
           let tabBarVC = storyboard.instantiateViewController(withIdentifier: "TabBarVC") as? TabBarViewController else { 
           return
 }`

VasckMe replied:
Fixed

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,95 @@
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        guard let tabBarVC = storyboard.instantiateViewController(withIdentifier: "TabBarVC") as? TabBarViewController else { return }
+        if let views = tabBarVC.viewControllers,
+           let profileVC = views[1] as? ProfileViewController,
```
what if there will be no `views[1]`?

VasckMe replied:
It was the test one, but I have already fixed it

VasckMe committed:
Fixes in code

- "Lbl" in "Label"
- if style
- guard style
- problem with views[1]
- encapsulation

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,125 @@
+            !password.isEmpty,
+            let email = emailTextField.text,
+            !email.isEmpty
+        {
```
use `guard` instead of `if`

KoalisMe commented on RegistrationApp/RegistrationApp/SignInControllers/SignInViewController.swift:
```diff
@@ -0,0 +1,125 @@
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        guard let tabBarVC =
+            storyboard.instantiateViewController(withIdentifier:
+                "TabBarVC") as? TabBarViewController else { return }
```
style is:
```
guard 
         let tabBarVC = storyboard.instantiateViewController(withIdentifier:"TabBarVC") as? TabBarViewController 
else { 
        return 
}
```

VasckMe committed:
Fixes

- Again changed guard style
- If changed on guard

progmev reviewed (approved):
+1
